[dnf5] argparser exit on 2

inknos opened the pull request.

lukash commented on microdnf/main.cpp:
```diff
@@ -120,6 +120,7 @@ inline RootCommand::RootCommand(libdnf::cli::session::Session & session) : Comma
 
 inline void RootCommand::run() {
     get_argument_parser_command()->help();
+    throw 2;
```
Oh no :grin:  Please don't throw an int :grin:  :skull_and_crossbones:

m-blaha commented on include/libdnf-cli/session.hpp:
```diff
@@ -75,7 +75,7 @@ class Command {
     /// Run the command. Must be overriden in inherited classes.
     /// @since 5.0
-    virtual void run() = 0;
+    virtual int run() = 0;
```
The return value should be documented in the comment.

inknos committed:
[dnf5] argparser return 2 on exception

dnf exits with error code 2 and then prints help
works for
- dnfdaemon-client
- dnfdaemon-client group
- microdnf
- microdnf group
- microdnf history
- microdnf module
